src/lib.rs

```diff
@@ -84,3 +126,97 @@ fn load_pem_certs(path: &Path) -> Result<Vec<CertificateDer<'static>>, Error> {
         })
         .collect()
 }
+
+/// Load certificate from certificate directory (what OpenSSL calls CAdir)
+///
+/// This directory can contain other files and directories. CAfile tends
+/// to be in here too. To avoid loading something twice or something that
+/// isn't an valid certificate. Load is limited to those files that have
+/// a hash-based file name matching the pattern used by OpenSSL. The hash
+/// is not verified, however.
+fn load_pem_certs_from_dir(dir: &Path) -> Result<Vec<CertificateDer<'static>>, Error> {
+    let dir_reader = fs::read_dir(dir)?;
+    let mut certs = Vec::new();
+    for entry in dir_reader {
+        let entry = entry?;
+        let path = entry.path();
+        let file_name = path
+            .file_name()
+            .expect("dir entry with no name");
```

Comment:
I think it might be preferable to continue the iteration rather than panic in this (unlikely) case. Or, alternatively, since this is already a fn returning a `Result` we could cook up a `Error` to yield here.

Reply:
I used `.expect()` here because I believe this to be unreachable. We are iterating over a directory and extracting `.file_name()`. As far as I know, there can be no file of any sort in a directory that doesn't have a name (except `"."` and `".."` which aren't yielded).

An alternative would be to use `entry.path()` which will yield an OsStr (not Option<OsStr>). It would mean an additional allocation but avoid the `.expect()`.

Let me know what you prefer.

Reply:
That sounds reasonable, a comment with this context would be an OK way to resolve this feedback if you'd like to keep the opportunity expect 👍

Reply:
Comment added